src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/spec/proceed/confirmation/PayImmediatelyConfText.java

```diff
@@ -27,7 +27,7 @@ public Optional<String> generateTextFor(CaseData caseData) {
         LocalDate whenBePaid = Optional.ofNullable(caseData.getRespondToClaimAdmitPartLRspec())
             .map(RespondToClaimAdmitPartLRspec::getWhenWillThisAmountBePaid)
             .orElse(null);
-        String formattedWhenBePaid = formatLocalDate(whenBePaid, DATE);
+        String formattedWhenBePaid = whenBePaid != null ? formatLocalDate(whenBePaid, DATE) : "";
```

Comment:
If `formattedWhenBePaid` will be empty then the message will be incorrect.
`They must make sure you have the money by %s. "`

Can you check if this can ever happen?